confluent-schema-manager/src/main/java/com/homeaway/streamingplatform/extensions/schema/ConfluentSchemaManager.java

```diff
@@ -0,0 +1,62 @@
+package com.homeaway.streamingplatform.extensions.schema;
+
+import com.google.common.base.Preconditions;
+import com.homeaway.streamingplatform.exceptions.SchemaException;
+import com.homeaway.streamingplatform.exceptions.SchemaManagerException;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG;
+
+@Slf4j
+public class ConfluentSchemaManager implements SchemaManager {
+
+    private SchemaRegistryClient schemaRegistryClient;
+
+    @Override
+    public SchemaReference registerSchema(String subject, String schema) throws SchemaManagerException {
+        org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schema);
+
+        SchemaReference schemaReference;
+        try {
+            long id = schemaRegistryClient.register(subject, avroSchema);
+            long version = schemaRegistryClient.getLatestSchemaMetadata(subject).getVersion();
+            schemaReference = new SchemaReference(subject, id, version);
+        } catch (IOException e) {
+            log.error("caught an IOException while registering a new schema for subject '{}'", subject);
+            throw new SchemaManagerException(e);
+        } catch (RestClientException e) {
+            log.error("caught a RestClientException while registering a new schema for subject '{}'", subject);
+            throw new SchemaManagerException(e);
+        }
```

Comment:
why differentiate ?

why not just :
```
} catch (RuntimeException e) {
...
}
```
is RestClientException checked exception?
If yes, then consider the folloing
```
} catch (RestClientException | RuntimeException exception) {...}
```

Reply:
It's two separate checked exceptions, neither of which inherit from the other so I changed it to `(IOException | RestClientException e)`

Reply:
ended up changing this to `catch (IOException | RestClientException | RuntimeException e) {` to include catching an unchecked exception thrown via registration failure